docs/recipes/PWA.md

```diff
@@ -0,0 +1,339 @@
+<!--ONE-DOCS-HIDE end-->
+
+[one-app-config]: ../api/modules/App-Configuration.md
+[frank-lloyd-root]: ../../prod-sample/sample-modules/frank-lloyd-root/0.0.3/README.md
+[one-service-worker]: https://github.com/americanexpress/one-service-worker
+
+# PWA
+
+## 📖 Table of Contents
+* [Overview](#overview)
+* [Setup](#setup)
+* [Offline Support](#offline-support)
+* [App Installation](#app-installation)
+
+## Overview
+
+In this compilation of mini recipes well be getting our feet wet with Progressive Web Apps
+(PWA) in One App. We'll learn a bit about the service worker and how to use the web
+manifest with One App. Each mini recipe below will cover UX issues and the technical
+solutions to refine your One App PWA.
+If any of these terms are new, we have links ready to get you familiar below in this section.
+We recommend going through and reviewing the links before using PWA if you haven't
+already done so.
+
+**Links**
+
+If you want to learn more about PWAs, service workers and web manifests,
+here are a collection of links that talk more about the technologies involved:
+
+- [PWA](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps)
+- [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API)
+- [Web App Manifest](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps)
+- [App Installs](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Developer_guide/Installing)
+- [Online/Offline](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events)
+- [Permissions](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API)
+- [Cache](https://developer.mozilla.org/en-US/docs/Web/API/Cache)
+- [Cache Storage](https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage)
+
+**Examples**
+
+* [`frank-lloyd-root` v0.0.3](frank-lloyd-root) sample module configures and enables PWA.
+
+## Setup
+
+Let's start by configuring One App to enable the service worker.
+
+**One App**
+
+First thing is we need to set `ONE_SERVICE_WORKER` before we run One App:
+
+```bash
+export ONE_SERVICE_WORKER=1 || true
+```
+
+This is set to `false` by default.
+
+**Root Module**
+
+Next step is to add it to your `appConfig` on your root module. The `pwa` key has
+a few options that we can use to turn on the service worker and create a web manifest.
+Here is a sample configuration:
+
+```jsx
+export default function RootModule({ children }) {
+  return children;
+}
+
+RootModule.childRoutes = () => [
+  <Route path="/" />,
+];
+
+RootModule.holocron = {
+  name: 'root-module',
+};
+
+if (!global.BROWSER) {
+  RootModule.appConfig = {
+    // along with other config
+    pwa: {
+      serviceWorker: true,
+      manifest: {
+        name: 'Pottery Store',
+        icons: [
+          {
+            src: 'https://drums.example.com/images/amphora-96.png',
+            sizes: '96x96',
+            type: 'image/png',
+          },
+        ],
+      },
+    },
+  };
+}
+```
+
```

Comment:
can you add a description of what this config will achieve

Reply:
I added comments to the config items and expanded the config (b912ca5)